drivers/sensor/microchip/mcp9600/mcp9600.c

```diff
@@ -64,23 +66,40 @@ static int mcp9600_sample_fetch(const struct device *dev, enum sensor_channel ch
 	uint8_t buf[2];
 	int ret;
 
-	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_THRMCP_TEMP &&
+	    chan != SENSOR_CHAN_THRMCP_COLD_JUNCTION_TEMP) {
 		LOG_ERR("Unsupported sensor channel");
 		return -ENOTSUP;
 	}
 
-	/* read signed 16 bit double-buffered register value */
-	ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, sizeof(buf));
-	if (ret < 0) {
-		data->temp = 1;
-		return ret;
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_THRMCP_TEMP) {
+		/* read signed 16 bit double-buffered register value */
+		ret = mcp9600_reg_read(dev, MCP9600_REG_TEMP_HOT, buf, sizeof(buf));
+		if (ret < 0) {
+			data->thermocouple_temp = 1;
+			return ret;
+		}
+
+		/* device's hot junction register is a signed int */
+		data->thermocouple_temp = (int32_t)(int16_t)(buf[0] << 8) | buf[1];
+
+		/* 0.0625C resolution per LSB */
+		data->thermocouple_temp *= 62500;
```

Comment:
It would be better if you use a macro for 62500 value instead of static usage. 
Also these value uses multiple time in the file to centralize it, it should be a macro.

Reply:
Sure, that makes sense.